jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java

```diff
@@ -575,746 +599,144 @@ public void createGlobalDirOnNewProject( @Observes NewProjectEvent newProjectEve
+                                    getBytesFromFile(new File(defaultCustomEditorsJSON)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default custom editors.");
+                }
+
+                // default color themes
+                try {
+                    String defaultColorThemesJSON = servletContext.getRealPath(designerContext + "/defaults/themes.json");
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "themes.json")),
+                                    getBytesFromFile(new File(defaultColorThemesJSON)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default color themes.");
+                }
+
+                // default icons
+                try {
+                    String defaultEmailIcon = servletContext.getRealPath(designerContext + "/defaults/defaultemailicon.gif");
+                    String defaultLogIcon = servletContext.getRealPath(designerContext + "/defaults/defaultlogicon.gif");
+                    String defaultServiceIcon = servletContext.getRealPath(designerContext + "/defaults/defaultservicenodeicon.png");
+
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "defaultemailicon.gif")),
+                                    getBytesFromFile(new File(defaultEmailIcon)));
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "defaultlogicon.gif")),
+                                    getBytesFromFile(new File(defaultLogIcon)));
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "defaultservicenodeicon.png")),
+                                    getBytesFromFile(new File(defaultServiceIcon)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default icons.");
+                }
+
+                // default patterns info
+                try {
+                    String defaultPatternsJSON = servletContext.getRealPath(designerContext + "/defaults/patterns.json");
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "patterns.json")),
+                                    getBytesFromFile(new File(defaultPatternsJSON)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default patterns.json");
+                }
+            } else {
+                logger.error("Null servlet context.");
+            }
         }
 
         ioService.endBatch();
     }
+
+    protected static byte[] getBytesFromFile(File file) throws java.io.IOException {
+        if (file == null || file.length() > Integer.MAX_VALUE) {
+            return null;
+        }
+
+        long length = file.length();
+        byte[] bytes;
+        InputStream is = null;
+        try {
```

Comment:
You can use try-with-resources statement and avoid the finally block alltogether..
```
try (InputStream is = new FileInputStream(file)) {
 ...
}
```

Reply:
Second thought: you can replace the entire method by `IOUtils.toByteArray(new FileInputStream(file));`